### ISSUE TITLE
Add how much space an app takes up on the user's device

### ISSUE THREAD

**Issue by ueman** [getsentry/sentry-cocoa#1664]
### Problem Statement

In the iOS settings (General -> Storage) I can see how much space an app takes up. This is split into:

- App Size
- Documents & Data

The use case is, that I can see if the app creates too much data or if I don't properly clean up data.

Relates to https://github.com/getsentry/sentry-java/issues/1917

**Comment by brustolin** [comment]
Documents e data size is interesting. We could use something similar to [this](https://github.com/NikolaiRuhe/NRFoundation/blob/master/NRFoundation/NRFileManager.m), but the solution is to traverse all files inside the app folder. If the app has created hundreds of files, this can be expensive. The sample I read says something about 35ms for 100 files.

To get available disk space we can use this snippet:

```swift 
let values = try url.resourceValues(forKeys: [.volumeAvailableCapacityForImportantUsageKey])
if let capacity = values.volumeAvailableCapacityForImportantUsage
```

**Comment by bruno-garcia** [comment]
It doesn't seem like feasible to attach either of those on each event because the overhead to retrieve this value "fresh" is rather expensive (depending on the amount of data on the folders). 

We had a brainstorming session for this value to go out as measurements. Not for `App Size` because it's easily accessible in the App Store.
But if we're reading "Documents & Data" every few minutes, close enough when we capture a transaction (we can read it right before capturing the transaction without blocking anything), this could be very valuable.

**Comment by kevinrenskers** [comment]
I am going to send this as `event.context.app.data_storage`, as an integer. There isn't an existing property for the app data size (see https://develop.sentry.dev/sdk/event-payloads/contexts/), so I made this key up myself. It's also not rendered nicely in a human readable format yet. Where can we ask the sentry.io team if this key makes sense, and if they can render this one nicely?

**Comment by bruno-garcia** [comment]
Please be explicit about the `unit` we're using here. If it's `bytes`, for example, we need to document this in code and on develop.sentry.dev.

Also, the UI should be able to render KB/MB/GB, etc, if the field is set to the right unit in the Sentry front end

**Comment by kevinrenskers** [comment]
It's in bytes like all the other memory and storage sizes we send

**Comment by philipphofmann** [comment]
Yes, please use the same approach as for these values https://github.com/getsentry/sentry-cocoa/blob/ceaa5790db2d26f454ad7962a9aaa62d09467449/Sources/Sentry/SentryCrashIntegration.m#L251-L253

They show up like this in Sentry:
<img width="314" alt="Screen Shot 2022-09-21 at 15 26 17" src="https://user-images.githubusercontent.com/2443292/191516081-4460dbe8-470d-4639-9d09-9cdbb4abf0d8.png">

We need changes in Sentry to support that. Similar to these lines of code https://github.com/getsentry/sentry/blob/0dbd18fc57a5a360fdfca921f59fbee4d0f650c6/static/app/components/events/contexts/device/getDeviceKnownDataDetails.tsx#L120-L127
```
case DeviceKnownDataType.EXTERNAL_STORAGE_SIZE: {
      return {
        subject: t('External Storage Size'),
        value: data.external_storage_size ? (
          <FileSize bytes={data.external_storage_size} />
        ) : undefined,
      };
    }
```

@priscilawebdev, can you maybe tell us how to add another known datatype in the file above?

**Comment by kevinrenskers** [comment]
> Yes, please use the same approach as for these values

I'm not sure what you mean with this. What approach do you mean? I'm setting an integer of bytes to the field `event.context.app.data_storage`. Even though it happens in a different file, the approach in the end is the same, right? Sentry's UI just needs to know about it.

**Comment by philipphofmann** [comment]
> Even though it happens in a different file, the approach in the end is the same, right? Sentry's UI just needs to know about it.

Yes, exactly.

**Comment by philipphofmann** [comment]
@ueman or @marandaneto, do you already send this for the dart SDK? If yes, which name did you choose for this property?

**Comment by ueman** [comment]
No, the Dart SDK doesn't send it.

**Comment by marandaneto** [comment]
nope, that would come directly from the iOS or Android SDK if possible.

**Comment by kevinrenskers** [comment]
We discussed this in our meeting today, and we're closing this issue for now until we can track metrics over time. We can't measure the Documents & Data size in real time for every single transaction because it's too slow, so it doesn't make sense to send this data as part of a transaction - it'll be stale data.

What would be interesting is to see the value change over time, for which metrics would be the way to go. This is something to pick up in the future.

**Comment by priscilawebdev** [comment]
> case DeviceKnownDataType.EXTERNAL_STORAGE_SIZE

To make this an official support field, we would need to:


**Relay**

Add the new field to Relay in [relay-event-schema/src/protocol/contexts/device.rs#L11](https://github.com/getsentry/relay/blob/b97e972470adb15ac5a479375752b2ede6adfd63/relay-event-schema/src/protocol/contexts/device.rs#L11)

**Frontend**

1 - Add to Enum in https://github.com/getsentry/sentry/blob/7015d14db676f1e22f256c202d1e1d38a136e65f/static/app/types/event.tsx#L391

`DATA_STORAGE = 'data_storage',`


2 - Add to the TypeScript interface in https://github.com/getsentry/sentry/blob/7015d14db676f1e22f256c202d1e1d38a136e65f/static/app/types/event.tsx#L434

`[DeviceContextKey.DATA_STORAGE]?: number;`


3 - Add Display Logic in https://github.com/getsentry/sentry/blob/7015d14db676f1e22f256c202d1e1d38a136e65f/static/app/components/events/contexts/knownContext/device.tsx#L131


```
case DeviceContextKey.DATA_STORAGE: {
  return {
    key: ctxKey,
    subject: t('Data Storage'),
    value: data.data_storage ? <FileSize bytes={data.data_storage} /> : undefined,
  };
}
```
